ci/ci-cppcheck.py

```
@@ -99,6 +102,46 @@ def find_platformio_ini_in_tree(base_dir: Path) -> Path | None:
+        f"--project={compile_db}",
+        "--enable=warning,performance,portability",
+        "--inline-suppr",
+        "--error-exitcode=1",
+        "--suppress=missingIncludeSystem",
+        # Limit analysis to src/ to match `pio check --src-filters=+<src/>`.
+        # cppcheck reads the DB's sources and filters post-hoc.
+        f"--file-filter={project_root / 'src'}/*",
+    ]
+    print(f"Running cppcheck against compile DB: {compile_db}")
+    print(f"  command: {' '.join(cmd)}")
+    cp = subprocess.run(cmd)
```

Ruff (0.15.10)

[error] `subprocess` call: check for execution of untrusted input

(S603)